File: app/src/main/java/com/example/a18/path/transition/AutoFragment.java
package com.example.a18.path.transition;

import android.support.transition.TransitionManager;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.a18.path.BaseFragment;
import com.example.a18.path.R;

import butterknife.BindView;
import butterknife.OnClick;


public class AutoFragment extends BaseFragment {

    @BindView(R.id.transitions_container)
    ViewGroup transitions_container;
    @BindView(R.id.text)
    TextView textView;
    private boolean visible;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_auto;
    }


    @OnClick(R.id.button)
    void button() {
        TransitionManager.beginDelayedTransition(transitions_container);
        textView.setVisibility((visible = !visible) ? View.VISIBLE : View.GONE);
    }
}

File: m_transition/src/main/java/pdog18/com/module_transition/AutoFragment.kt
package pdog18.com.module_transition

import android.os.Bundle
import androidx.transition.TransitionManager
import android.view.View
import kotlinx.android.synthetic.main.fragment_auto.*
import pdog18.com.base.BaseFragment
import pdog18.com.base.Layout

@Layout(R.layout.fragment_auto)
class AutoFragment : BaseFragment() {

    private var visible: Boolean = false

    override fun getLayoutId(): Int {
        return R.layout.fragment_auto
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        button.setOnClickListener {
            TransitionManager.beginDelayedTransition(transitions_container)
            visible = !visible
            text.visibility = if (visible) View.VISIBLE else View.GONE
        }
    }
}
